RemoveDCOffset.cpp

```diff
@@ -0,0 +1,355 @@
+      double trackStart = track->GetStartTime();
+      double trackEnd = track->GetEndTime();
+
+      // Set the current bounds to whichever left marker is
+      // greater and whichever right marker is less:
+      mCurT0 = std::max(trackStart, mT0);
+      mCurT1 = std::min(trackEnd, mT1);
+
+      // Process only if the right marker is to the right of the left marker
+      if (mCurT1 > mCurT0) {
+         wxString trackName = track->GetName();
+
+         std::vector<float> extents;
+         float maxExtent{ std::numeric_limits<float>::lowest() };
+         std::vector<float> offsets;
+
+         const auto channels = track->Channels();
+         // mono or 'stereo tracks independently'
+         const bool oneChannel = (channels.size() == 1);
+         auto msg = oneChannel
+            ? topMsg +
+               XO("Analyzing: %s").Format(trackName)
+            : topMsg +
+               // TODO: more-than-two-channels-message
+               XO("Analyzing first track of stereo pair: %s").Format(trackName);
+
+         const auto progressReport = [&](double fraction){
+            return !TotalProgress(
+               (progress + fraction / double(2 * GetNumWaveTracks())), msg);
+         };
+
+         // Analysis loop over channels collects offsets and extent
+         for (auto channel : channels) {
+            float offset = 0;
+            float extent = 0;
+            bGoodResult = AnalyseTrack(*channel, progressReport,
+               mCurT0, mCurT1, offset, extent);
+            if (!bGoodResult)
+               goto break2;
+            progress += 1.0 / double(2 * GetNumWaveTracks());
+            extents.push_back(extent);
+            maxExtent = std::max(maxExtent, extent);
+            offsets.push_back(offset);
+            // TODO: more-than-two-channels-message
+            if (!oneChannel)
+               msg = topMsg +
+                  XO("Analyzing second track of stereo pair: %s")
+                     .Format(trackName);
+         }
+
+         if (oneChannel) {
+            if (TrackList::NChannels(*track) == 1)
+               // really mono
+               msg = topMsg +
+                  XO("Processing: %s").Format(trackName);
+            else
+               //'stereo tracks independently'
+               // TODO: more-than-two-channels-message
+               msg = topMsg +
+                  XO("Processing stereo channels independently: %s")
+                     .Format(trackName);
+         }
+         else
+            msg = topMsg +
+               // TODO: more-than-two-channels-message
+               XO("Processing first track of stereo pair: %s")
+                  .Format(trackName);
+
+         // Use multiplier in the second, processing loop over channels
+         auto pOffset = offsets.begin();
+         auto pExtent = extents.begin();
+         for (const auto channel : channels) {
+            const auto extent = oneChannel ? *pExtent++: maxExtent;
```

Comment:
Hey Patrick, thanks for your contribution!
It looks like you don't use `extent`, in the end.